### PR TITLE
Add transmission start notification

### PR DESCRIPTION
**Pull request by hubertmis**
This PR adds two optional features:

1.  Notification to MAC layer that transmission of data frame or Ack started
2. Included timestamps in notifications of received data frame or Ack

### CODE REVIEW

File: src/nrf_drv_radio802154.h [hunk 1]
```diff
@@ -509,6 +570,66 @@ extern void nrf_drv_radio802154_transmitted(const uint8_t * p_frame,
...
+ * @note Including timestamp to received frame uses resources like CPU time and memory. If timestamp
+ *       is not required use @sa nrf_drv_radio802154_received instead.
+ *
+ * @param[in]  p_frame  Pointer to buffer containing PSDU of transmitted frame.
+ * @param[in]  p_ack    Pointer to buffer containing received ACK payload (PHR excluding FCS).
+  *                     If ACK was not requested @p p_ack is set to NULL.
+ * @param[in]  length   Length of received ACK payload.
+ * @param[in]  power    RSSI of received frame or 0 if ACK was not requested.
+ * @param[in]  lqi      LQI of received frame or 0 if ACK was not requested.
+ * @param[in]  time     Timestamp taken when received last symbol of ACK of 0 if ACK was not
```
Comment (rlubos) [inline src/nrf_drv_radio802154.h]: ... **or** 0 if ACK was not requested.

File: src/nrf_drv_radio802154_config.h [hunk 2]
```diff
@@ -366,6 +366,42 @@ extern "C" {
...
+ ******************************************************************************/
+
+/**
+ * @def NRF_DRV_RADIO802154_TX_STARTED_NOTIFY_ENABLED
+ *
+ * If notification of started transmission should be enabled in the driver.
+ *
+ */
+#ifndef NRF_DRV_RADIO802154_TX_STARTED_NOTIFY_ENABLED
+#if NRF_DRV_RADIO802154_ACK_TIMEOUT_ENABLED ||                                                     \
```
Comment (rlubos) [inline src/nrf_drv_radio802154_config.h]: I'd add some explaination that default behavior depends on those configs.

File: src/nrf_drv_radio802154_config.h [hunk 3]
```diff
@@ -366,6 +366,42 @@ extern "C" {
 #define NRF_DRV_RADIO802154_ACK_TIMEOUT_DEFAULT_TIMEOUT 7000
 #endif
 
+/*******************************************************************************
+ * @section Transmission start notification feature configuration.
```
Comment (rlubos) [inline src/nrf_drv_radio802154_config.h]: Do these configs need separate sections? Couldn't they fit into radio section?

Comment (hubertmis) [reply src/nrf_drv_radio802154_config.h]: `...TX_STARTED_NOTIFY_ENABLED` must be defined after `...CSMA_CA...` and `...ACK_TIMEOUT...`. But I can move `...FRAME_TIMESTAMP_ENABLED` to radio driver configuration.

Comment (hubertmis) [reply src/nrf_drv_radio802154_config.h]: Hm, must it? Perhaps macro can be defined after it is used.

Comment (hubertmis) [reply src/nrf_drv_radio802154_config.h]: No. It must be defined after. Ack timeout does not work instead.

**Pull request merged**